### PR TITLE
feat(e2e-connector-test): CI FE-499

### LINKED ISSUE #243: Create e2e test action on CI
Run tests on local AND testnet

Get testnet pk with funds

### PR DESCRIPTION
**Pull request by nelitow**
- Closes #243 

This PR implements CI workflows for the e2e connector tests. It builds on work done it https://github.com/FuelLabs/fuel-connectors/pull/199 

Actual tests are implemented there, thats why the outdates tests were disabled here, so that I can update there with the new file structure without tests failing here.
https://github.com/FuelLabs/fuel-connectors/pull/239

### CODE REVIEW

File: examples/react-app/tests/FuelConnector.test.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  test('should render connect button', async ({ page }) => {
+    await page.goto('http://localhost:5173/');
+    await page.waitForSelector('text=Connect Wallet');
+    await page.waitForSelector('text=Connect Wallet:enabled');
+
+    // click connect button
+    await page.click('text=Connect Wallet');
+
+    // should see modal with Fuel Wallet button
+    await page.waitForSelector('text=Fuel Wallet');
```
Comment (LuizAsFight) [inline examples/react-app/tests/FuelConnector.test.ts:14]: this test should not exist as they are already in the e2e-test folder

File: e2e-tests/react-app/utils/connect.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+export const connect = async (
+  page: Page,
+  fuelWalletTestHelper: FuelWalletTestHelper,
+  walletName = 'Fuel Wallet',
+) => {
+  const connectButton = getButtonByText(page, 'Connect');
+  await connectButton.click();
+  await getByAriaLabel(page, `Connect to ${walletName}`, true).click();
+  await fuelWalletTestHelper.walletConnect();
+};
```
Comment (LuizAsFight) [inline e2e-tests/react-app/utils/connect.ts:14]: this file dont need to exist

**Review by LuizAsFight — CHANGES_REQUESTED** [review]
this PR should only include CI implementation, but instead its including orthogonal changes and changes made in other PRs already

File: .github/workflows/pr-tests.yml [hunk 3]
```diff
@@ -76,3 +76,33 @@ jobs:
...
+        run: pnpm install
+        working-directory: ./e2e-tests
+
+      - name: Install Playwright browsers
+        run: pnpm exec playwright install
+        working-directory: ./e2e-tests
+
+      - name: Run Playwright tests
+        run: pnpm exec playwright test
+        working-directory: ./e2e-tests
```
Comment (LuizAsFight) [inline .github/workflows/pr-tests.yml:108]: - no needs to run pnpm install
- pnpm can be installed in setup-node
- install playwright with chromiun
- execute tests from script on root
- execute using "xvfb-run --auto-servernum -- " due to linux issues

```suggestion
      - name: Set up Node.js
        uses: actions/setup-node@v3
        with:
          node-version: 20.11.0
          pnpm-version: 8.9.2

      - name: Install Playwright browsers
        run: pnpm exec playwright install  --with-deps chromium
        working-directory: ./e2e-tests

      - name: Run Playwright tests
        run: xvfb-run --auto-servernum -- pnpm test:e2e
        working-directory: ./e2e-tests
```

File: .github/actions/e2e-tests-connectors/action.yaml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+      with:
+        name: playwright-report
+        path: |
+          packages/app/playwright-report/
+          packages/app/playwright-html/
+        retention-days: 30
+
+    - name: Stop Test Node
+      shell: bash
+      run: pnpm node:clean
```
Comment (LuizAsFight) [inline .github/actions/e2e-tests-connectors/action.yaml:49]: this file should not exist right?

**Review by LuizAsFight — CHANGES_REQUESTED** [review]

**Comment by nelitow** [comment]
@LuizAsFight please provide the secrets if available for the mnemonic as `VITE_MASTER_WALLET_MNEMONIC` was returning empty.

**Pull request closed without merging**